lib/spack/spack/package.py

```diff
@@ -1786,7 +1789,6 @@ def check_for_unfinished_installation(
 
         if restage and self.stage.managed_by_spack:
             self.stage.destroy()
-            self.stage.create()
```

Comment:
Why is this line removed?

Reply:
Because the stage destruction occurs before the stage should be created. Failed installations using `spack install --cache-only` were creating a stage directory at this point, and there's no reason to create the directory yet. We don't even know yet whether we need it. So it's better to only destroy it now, and recreate it when we know whether we need it.